File: Components/Util/Serialization.vb
Imports System.Xml.Serialization
Imports System.Text
Imports System.Xml
Imports System.IO

Namespace DNNStuff.SQLViewPro

    Public Class Serialization

#Region " XML Serialization Support"
        Private Shared Function UTF8ByteArrayToString(ByVal characters() As [Byte]) As [String]
            Dim encoding As New UTF8Encoding
            Dim constructedString As [String] = encoding.GetString(characters)
            Return constructedString
        End Function 'UTF8ByteArrayToString
        Private Shared Function StringToUTF8ByteArray(ByVal pXmlString As [String]) As [Byte]()
            Dim encoding As New UTF8Encoding
            Dim byteArray As [Byte]() = encoding.GetBytes(pXmlString)
            Return byteArray
        End Function 'StringToUTF8ByteArray

        Public Shared Function SerializeObjectOld(ByVal o As Object, ByVal t As Type) As String
            Try
                Dim ms As New MemoryStream
                Dim xs As New XmlSerializer(t)
                Dim writer As New XmlTextWriter(ms, New System.Text.UTF8Encoding)
                writer.Formatting = Formatting.Indented

                xs.Serialize(writer, o)
                ms = CType(writer.BaseStream, MemoryStream)
                Return UTF8ByteArrayToString(ms.ToArray())
            Catch e As Exception
                Return Nothing
            End Try
        End Function 'SerializeObject

        Public Shared Function SerializeObject(ByVal o As Object, ByVal t As Type) As String
            Try
                '                Dim ms As New MemoryStream
                Dim sw As New UTF8StringWriter()
                Dim xs As New XmlSerializer(t)

                Dim settings As New XmlWriterSettings()
                settings.NewLineHandling = NewLineHandling.Entitize
                settings.Indent = True
                settings.Encoding = New UTF8Encoding()
                settings.IndentChars = " "

                Dim writer As XmlWriter = XmlWriter.Create(sw, settings)
                xs.Serialize(writer, o)

                Return sw.ToString()

                'ms = CType(writer.BaseStream, MemoryStream)
                'Return UTF8ByteArrayToString(ms.ToArray())
            Catch e As Exception
                Return Nothing
            End Try
        End Function 'SerializeObject

        Public Shared Function DeserializeObjectOld(ByVal s As String, ByVal t As Type) As Object
            Dim xs As New XmlSerializer(t)
            Dim ms As New MemoryStream(StringToUTF8ByteArray(s))
            Return xs.Deserialize(ms)
        End Function 'DeserializeObject

        Public Shared Function DeserializeObject(ByVal s As String, ByVal t As Type) As Object
            Dim sr As New StringReader(s)
            Dim xs As New XmlSerializer(t)

            Dim settings As New XmlReaderSettings()
            Dim xr As XmlReader = XmlReader.Create(sr, settings)
            Return xs.Deserialize(xr)
        End Function 'DeserializeObject

        Friend Class UTF8StringWriter
            Inherits StringWriter
            Public Overrides ReadOnly Property Encoding As System.Text.Encoding
                Get
                    Return Encoding.UTF8
                End Get
            End Property
        End Class

#End Region

    End Class
End Namespace

File: Components/Util/Serialization.cs


using System;


using System.Xml.Serialization;
using System.Text;
using System.Xml;
using System.IO;


namespace DNNStuff.SQLViewPro
{
	
	public class Serialization
	{
		
#region  XML Serialization Support
		private static string UTF8ByteArrayToString(byte[] characters)
		{
			var encoding = new UTF8Encoding();
			var constructedString = encoding.GetString(characters);
			return constructedString;
		} //UTF8ByteArrayToString
		private static byte[] StringToUTF8ByteArray(string pXmlString)
		{
			var encoding = new UTF8Encoding();
			var byteArray = encoding.GetBytes(pXmlString);
			return byteArray;
		} //StringToUTF8ByteArray
		
		public static string SerializeObjectOld(object o, Type t)
		{
			try
			{
				var ms = new MemoryStream();
				var xs = new XmlSerializer(t);
				var writer = new XmlTextWriter(ms, new UTF8Encoding());
				writer.Formatting = Formatting.Indented;
				
				xs.Serialize(writer, o);
				ms = (MemoryStream) writer.BaseStream;
				return UTF8ByteArrayToString(ms.ToArray());
			}
			catch (Exception)
			{
				return null;
			}
		} //SerializeObject
		
		public static string SerializeObject(object o, Type t)
		{
			try
			{
				//                Dim ms As New MemoryStream
				var sw = new UTF8StringWriter();
				var xs = new XmlSerializer(t);
				
				var settings = new XmlWriterSettings();
				settings.NewLineHandling = NewLineHandling.Entitize;
				settings.Indent = true;
				settings.Encoding = new UTF8Encoding();
				settings.IndentChars = " ";
				
				var writer = XmlWriter.Create(sw, settings);
				xs.Serialize(writer, o);
				
				return sw.ToString();
				
				//ms = CType(writer.BaseStream, MemoryStream)
				//Return UTF8ByteArrayToString(ms.ToArray())
			}
			catch (Exception)
			{
				return null;
			}
		} //SerializeObject
		
		public static object DeserializeObjectOld(string s, Type t)
		{
			var xs = new XmlSerializer(t);
			var ms = new MemoryStream(StringToUTF8ByteArray(s));
			return xs.Deserialize(ms);
		} //DeserializeObject
		
		public static object DeserializeObject(string s, Type t)
		{
			var sr = new StringReader(s);
			var xs = new XmlSerializer(t);
			
			var settings = new XmlReaderSettings();
			var xr = XmlReader.Create(sr, settings);
			return xs.Deserialize(xr);
		} //DeserializeObject
		
		internal class UTF8StringWriter : StringWriter
		{
			public override Encoding Encoding => Encoding.UTF8;
		}
		
#endregion
		
	}
}

